tests/connection_test.py

```diff
@@ -33,6 +33,13 @@ def test_connect_tcp(request, create_connection, loop, server):
 
 
 @pytest.mark.run_loop
+def test_connect_tcp_timeout(request, create_connection, loop, server):
+    with pytest.raises(asyncio.TimeoutError):
+        yield from create_connection(
+            server.tcp_address, loop=loop, timeout=0)
```

Comment:
Regarding tests, I think we can patch `asyncio.open_connection` with `mock`, something like this:
```python
@mock.patch('aioredis.connection.asyncio.open_connection')
def test(open_conn_mock):
    open_conn.side_effect = lambda *a, **k: asyncio.sleep(longer_than_timeout)
    ...
```